src/ape/utils/misc.py

```diff
@@ -230,6 +231,26 @@ def inner(*args, **kwargs):
     return inner
 
 
+class cached_iterator(property):
+    """
+    A cached iterator decorator. Use it as you would ``@cached_property``.
+    The intent is to prevent repeated fetches of data.
+    """
+
+    cache: Optional[Tuple[Iterator[Any], ...]] = None
```

Comment:
Here this acts as a class variable, which probably isn't what you want it to do. Just move this line to `__init__` or use `hasattr` in `__get__`